deploy-board/deploy_board/static/js/deploy-board.js

```diff
@@ -89,6 +91,57 @@ function getDefaultPlacement(capacityCreationInfo) {
         return items.sort(function (item1, item2) { return item1.capacity < item2.capacity; })
     }
 
+    function determineColorClass(capacity) {
+        if (capacity < 50) {
+            return 'text-danger';
+        }
+        else if (capacity < 200) {
+            return 'text-warning';
+        }
+        return 'text-primary';
+    }
+
+    function getAZ(abstractName, azRegex) {
+        var matched = abstractName.match(azRegex);
+        return matched ? matched[0] : 'unknown';
+    }
+
+    function convertToPlacementOptionsAdv(placements) {
+        var options = {}
+        if (placements === null || placements.length < 1 ) {
+            return options;
+        }
+        const cellName = placements ? placements[0].cell_name : '1';
+        const cellNum = cellName[cellName.length - 1];
+        const azRegex = new RegExp(`\\b(${cellNum}[a-g])\\b`);
+
+        for (const placement of placements) {
+            var obj = {
+                value: placement.id,
+                text: `${placement.provider_name} | cap.: ${placement.capacity} | ${placement.abstract_name}`,
+                isSelected: placement.isSelected,
+                colorClass: determineColorClass(placement.capacity)
+            };
+            const group = `${getAZ(placement.abstract_name, azRegex)} (Parsed info, for reference only)`;
+            if ( !(group in options)) {
+                options[group] = [];
+            }
+            options[group].push(obj);
+        }
+        return options;
+    }
+
+    function convertToPlacementOptions(placements) {
+        return {'Basic Settings': placements};
+    }
+
+    function inDefault(item) {
+        const searchSource = item.assign_public_ip ? topCmpPublicIPPlacements : topCmpPrivateIPPlacements;
```

Comment:
Does this mean the checkbox decides the default subnets or does this affect search visibility?

Reply:
`assign_public_ip` filed of a placement object is stored in the DB and indicates whether the placement is a private or a public subnet. It's different from the UI button "Assign Public IP".